improver_tests/threshold/conftest.py

```diff
@@ -0,0 +1,295 @@
+    """Return a template for the name of a thresholded cube taking into
+    account the comparison operator and the application of vicinity
+    processing.
+
+    Args:
+        comparison_operator:
+            The comparator used for thresholding, e.g. >.
+        vicinity:
+            A vicinity radius if set, or None. Used to determine whether
+            the diagnostic name should include the vicinity term.
+    Returns:
+        The expected diagnostic name after thresholding.
+    """
+
+    vicinity_name = "_in_vicinity" if vicinity is not None else ""
+
+    if "g" in comparison_operator.lower() or ">" in comparison_operator:
+        return f"probability_of_{{cube_name}}{vicinity_name}_above_threshold"
+    else:
+        return f"probability_of_{{cube_name}}{vicinity_name}_below_threshold"
+
+
+@pytest.fixture
+def expected_result(
+    expected_single_value: float,
+    expected_multi_value: List[float],
+    collapse: bool,
+    comparator: str,
+    default_cube: Cube,
+) -> np.ndarray:
+    """Return the expected values following thresholding.
+    The default cube has a 0.5 value at location (2, 2), with all other
+    values set to 0. The expected result, for a ">" or ">=" threshold
+    comparator, is achieved by placing the expected value, or values for
+    multi-realization data without coordinate collapsing, at the (..., 2, 2)
+    location in an array full of zeros. This array matches the size of
+    the input cube.
+
+    If the comparator is changed to be be "<" or "<=" then the array is
+    filled with ones, and the expected value(s) is subtracted from 1 prior
+    to being places at the (..., 2, 2) location.
+
+    This function does no more than this to avoid making the tests harder
+    to follow. Cases for which the threshold value is equal to any of
+    the data values (without fuzziness) meaning that the ">" and ">=", or
+    "<" and "<=" comparators would give different results are handled
+    separately.
+
+    Args:
+        expected_single_value:
+            The value expected if a single realization or deterministic
+            input cube is being tested.
+        expected_multi_value:
+            The values expected for each realization is a multi-realization
+            input cube is being tested. If the collapse argument is true the
+            expected result is the mean of these values.
+        collapse:
+            Whether the test includes the collapsing of the realization
+            coordinate to calculate the final result.
+        comparator:
+            The comapartor being applied in the thresholding, either ">",
+            ">=", "<", or "<=".
+        default_cube:
+            The input cube to the test.
+    Returns:
+        A data array that contains the expected values after thresholding.
+    """
+
+    try:
+        n_realizations = default_cube.coord("realization").shape[0]
+    except CoordinateNotFoundError:
+        n_realizations = 1
+
+    if "l" in comparator:
```

Comment:
This appears to expect a comparator of `"le"` or `"lt"` while the doc-string says only `"<"` or `"<="` are expected.